tlsutil/generate.go

```diff
@@ -29,23 +29,7 @@ func GenerateSerialNumber() (*big.Int, error) {
 
 // GeneratePrivateKey generates a new ecdsa private key
 func GeneratePrivateKey() (crypto.Signer, string, error) {
-	pk, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		return nil, "", fmt.Errorf("error generating private key: %s", err)
-	}
-
-	bs, err := x509.MarshalECPrivateKey(pk)
-	if err != nil {
-		return nil, "", fmt.Errorf("error generating private key: %s", err)
-	}
-
-	var buf bytes.Buffer
-	err = pem.Encode(&buf, &pem.Block{Type: "EC PRIVATE KEY", Bytes: bs})
-	if err != nil {
-		return nil, "", fmt.Errorf("error encoding private key: %s", err)
-	}
-
-	return pk, buf.String(), nil
+	return connect.GeneratePrivateKey()
```

Comment:
Great job DRYing up this function!

Given the TODO above, shouldn't we replace it with a call to `GeneratePrivateKeyWithConfig` though :D?

Reply:
We should. But first we'd have to get the values from config and pass them all the way through to this function. Future enhancement maybe?